hail/src/main/scala/is/hail/utils/FlipbookIterator.scala

```diff
@@ -84,6 +86,65 @@ object FlipbookIterator {
     StagingIterator(sm)
 
   def empty[A] = StagingIterator(StateMachine.terminal[A])
+
+  def multiZipJoin[A: ClassTag](
+    its: IndexedSeq[FlipbookIterator[A]],
```

Comment:
For performance reasons, you probably want this to be an array.  IndexedSeq has several implementations and the default is often Vector, which you never want and has terrible performance characteristics.